music_assistant/providers/audioscrobbler/__init__.py

```
@@ -0,0 +1,325 @@
+) -> ProviderInstanceType:
+    """Initialize provider(instance) with given configuration."""
+    provider = ScrobblerProvider(mass, manifest, config)
+    pylast.logger.setLevel(provider.logger.level)
+    if provider.logger.level == logging.DEBUG:
+        # httpcore is quite spammy without providing useful information 99% of the time
+        logging.getLogger("httpcore").setLevel(logging.INFO)
+
+    return provider
+
+
+class ScrobblerProvider(PluginProvider):
+    """Plugin provider to support scrobbling of tracks."""
+
+    _network: pylast._Network = None
+    _currently_playing: str = None
+    _processing_queue_update = False
+
+    def _get_network_config(self) -> dict[str, ConfigValueType]:
+        return {
+            CONF_API_KEY: self.config.get_value(CONF_API_KEY),
+            CONF_API_SECRET: self.config.get_value(CONF_API_SECRET),
+            CONF_PROVIDER: self.config.get_value(CONF_PROVIDER),
+            CONF_USERNAME: self.config.get_value(CONF_USERNAME),
+            CONF_SESSION_KEY: self.config.get_value(CONF_SESSION_KEY),
+        }
+
+    async def loaded_in_mass(self) -> None:
+        """Call after the provider has been loaded."""
+        await super().loaded_in_mass()
+
+        if not self.config.get_value(CONF_SESSION_KEY):
+            self.logger.info("No session key available")
+            return
+
+        self._network = _get_network(self._get_network_config())
+
+        def wrap(cb: Callable[[MassEvent], None]):
+            async def handler(event: MassEvent) -> None:
+                try:
+                    await cb(event)
+                except Exception as e:
+                    self.logger.exception(e)
+
+            return handler
+
+        # subscribe to internal events
+        self.mass.subscribe(wrap(self._on_mass_queue_updated), EventType.QUEUE_UPDATED)
+        self.mass.subscribe(wrap(self._on_mass_media_item_played), EventType.MEDIA_ITEM_PLAYED)
```

I'm not a fan of this wrapping of event logic tbh
The core is already creating tasks for each event handler. 
Please just subscribe a callback as intended

---

Ah yeah, I forgot to add a comment about it. I'm not a fan either, so I'm happy to remove it.
I added it because I had some trouble with silent exceptions, which was very frustrating.

---

hmmm, exceptions should never go silent as we have a global catch-all exception handler for uncaught ones.  
I'll have to look into that, maybe a bug.

---

I don't know if there's a different in Python, but among others I ran into `method not found` kind of exceptions that didn't get logged. I imported `is_track` within the `TYPE_CHECKING` block, which (obviously) didn't work.

---

these should be catched by the catch-all exception handler. Are you running the debugger in vscode according to the predefined settings?